django/contrib/admin/views/autocomplete.py

```diff
@@ -24,11 +24,17 @@ def get(self, request, *args, **kwargs):
             return JsonResponse({'error': '403 Forbidden'}, status=403)
 
         self.term = request.GET.get('term', '')
+        self.paginator_class = self.model_admin.paginator
         self.object_list = self.get_queryset()
         context = self.get_context_data()
+
+        # specified ForeignKey by `AutocompleteMixin.get_url`
+        # ForeignKey may not be a primary_key
+        to_field = request.GET.get('_to_field', 'pk')
+
         return JsonResponse({
             'results': [
-                {'id': str(obj.pk), 'text': str(obj)}
+                {'id': str(getattr(obj, to_field)), 'text': str(obj)}
```

Comment:
`getattr` will throw a `ValueError` if the `to_field` does not exist, this has to be handled.

Reply:
Fixed in 99edcdc